pkg/provider/metric_selector_test.go

```
@@ -0,0 +1,43 @@
+	sel, err := parseMetricSelection(`{"metric_name": "", "reason": "no configured metric is relevant to this task"}`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if sel.MetricName != "" {
+		t.Errorf("got %+v, want empty metric_name to mean \"none chosen\"", sel)
+	}
+}
+
+func TestMockMetricSelectorReturnsConfiguredChoice(t *testing.T) {
+	m := &MockMetricSelector{Choice: "checkout_p95_latency"}
+	got, err := m.SelectMetric(nil, "speed up checkout", []MetricOption{{Name: "checkout_p95_latency"}})
```

golangci-lint (2.12.2)

[error] SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)